flexmeasures/data/models/reporting/tests/test_tibber_reporter.py

```diff
@@ -0,0 +1,252 @@
+            end=str(end),
+            tb_query_config=[
+                dict(sensor=EB.id, **tb_query_config_extra),
+                dict(sensor=BWV.id, **tb_query_config_extra),
+                dict(sensor=tibber_tariff.id, **tb_query_config_extra),
+                dict(sensor=da_prices.id, **tb_query_config_extra),
+            ],
+            transformations=[
+                dict(
+                    df_input="sensor_1",
+                    df_output="BWV",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
+                dict(method="add", args=[1]),  # this is to get 1 + BWV
+                dict(
+                    df_input="sensor_2",
+                    df_output="EB",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
+                dict(
+                    df_input="sensor_3",
+                    df_output="tibber_tariff",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
+                dict(
+                    df_input="sensor_4",
+                    df_output="da_prices",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
+                dict(
+                    method="multiply",
+                    args=[1 / 1000],
+                ),
+                dict(
+                    df_output="energy_price",
+                    df_input="EB",
+                    method="add",
+                    args=["@tibber_tariff"],
+                ),
+                dict(method="add", args=["@da_prices"]),
+                dict(method="multiply", args=["@BWV"]),
+            ],
+            final_df_output="energy_price",
+        )
+
+        super().__init__(reporter_config_raw)
+
+
+def beliefs_from_timeseries(index, values, sensor, source):
+    beliefs = []
+    for dt, value in zip(index, values):
+        beliefs.append(
+            TimedBelief(
+                event_start=dt,
+                belief_horizon=timedelta(hours=24),
+                event_value=value,
+                sensor=sensor,
+                source=source,
+            )
+        )
+
+    return beliefs
+
+
+@pytest.fixture()
+def tibber_test_data(fresh_db, app):
+    db = fresh_db
+
+    tax = GenericAssetType(name="Tax")
+    price = GenericAssetType(name="Price")
+
+    db.session.add_all([tax, price])
+
+    # Belastingdienst
+
+    electricity_price = GenericAsset(name="Electricity Price", generic_asset_type=price)
+
+    VAT = GenericAsset(name="VAT", generic_asset_type=tax)
+
+    electricity_tax = GenericAsset(name="Energy Tax", generic_asset_type=tax)
+
+    db.session.add_all([electricity_price, VAT, electricity_tax])
+
+    # Taxes
+    BWV = Sensor("BWV", generic_asset=VAT, event_resolution=timedelta(days=365))
+    EB = Sensor(
+        "EB", generic_asset=electricity_tax, event_resolution=timedelta(days=365)
+    )
```

Comment:
Missing sensor units, which defaults to unitless (encoded by timely-beliefs as an empty string).

Reply:
I've added the units.